### PR TITLE
[Merged by Bors] - chore: Reorganize results about `rank` and `finrank`.

### PR DESCRIPTION
**Pull request by erdOne**
The files `Mathlib.LinearAlgebra.FreeModule.Rank`, `Mathlib.LinearAlgebra.FreeModule.Finite.Rank`, `Mathlib.LinearAlgebra.Dimension` and `Mathlib.LinearAlgebra.Finrank` were reorganized into a
folder `Mathlib.LinearAlgebra.Dimension`, containing the following files
- `Basic.lean`: Contains the definition of `Module.rank`.
- `Finrank.lean`: Contains the definition of `FiniteDimensional.finrank`.
- `StrongRankCondition.lean`: Contains results about `rank` and `finrank`
      over rings satisfying strong rank condition
- `Free.lean`: Contains results about `rank` and `finrank` of free modules
- `Finite.lean`: Contains conditions or consequences for `rank` to be finite or zero
- `Constructions.lean`: Contains the calculation of the `rank` of various constructions.
- `DivisionRing.lean`: Contains results about `rank` and `finrank` of spaces over division rings.
- `LinearMap.lean`: Contains results about `LinearMap.rank`

API changes:
`IsNoetherian.rank_lt_aleph0` and `FiniteDimensional.rank_lt_aleph0` are replaced with
`rank_lt_aleph0`.
`Module.Free.finite_basis` was renamed to `Module.Finite.finite_basis`.
`FiniteDimensional.finrank_eq_rank` was renamed to `finrank_eq_rank`.
`rank_eq_cardinal_basis` and `rank_eq_cardinal_basis'` were removed 
in favour of `Basis.mk_eq_mk` and `Basis.mk_eq_mk''`.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/LinearAlgebra/Dimension/Basic.lean [hunk 1]
```diff
@@ -0,0 +1,375 @@
...
+  at most that of the target.
+* `LinearMap.rank_le_of_surjective`: the target of a surjective linear map has dimension
+  at most that of that source.
+
+## Implementation notes
+
+Many theorems in this file are not universe-generic when they relate dimensions
+in different universes. They should be as general as they can be without
+inserting `lift`s. The types `V`, `V'`, ... all live in different universes,
+and `V₁`, `V₂`, ... all live in the same universe.
```
Comment (jcommelin) [inline Mathlib/LinearAlgebra/Dimension/Basic.lean:32]: Does this still apply to this file? I guess at least `s/V/M/`?

File: Mathlib/LinearAlgebra/Dimension/Finite.lean [hunk 2]
```diff
@@ -0,0 +1,518 @@
...
+# Conditions for rank to be finite
+
+Also contains characterization for when rank equals zero or rank equals one.
+
+-/
+
+
+noncomputable section
+
+universe u v v' v'' u₁' w w'
```
Comment (jcommelin) [inline Mathlib/LinearAlgebra/Dimension/Finite.lean:24]: Do we need all of these? I think at least `u₁'` can go.

**Review by jcommelin — APPROVED** [review]
Thanks a lot for this effort :tada:

Two minor comments.

bors d+

**Comment by erdOne** [comment]
bors merge

**Comment by erdOne** [comment]
bors merge

**Pull request closed without merging**